app/src/test/java/com/hifeful/mealmania/presentation/MealManiaViewModelTest.kt

```diff
@@ -0,0 +1,121 @@
+package com.hifeful.mealmania.presentation
+
+import android.net.Uri
+import android.support.test.runner.AndroidJUnit4
+import com.hifeful.mealmania.presentation.util.DeeplinkResult
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+@RunWith(AndroidJUnit4::class)
+class MealManiaViewModelTest {
+    private val viewModel = MealManiaViewModel()
+
+    @Test
+    fun `Given mymeals page deeplink, When handleDeeplink is called, Then ShowMyMealsPage DeeplinkResult should be returned`() {
```

Comment:
I would suggest using the following template:
`with mymeals page deeplink - on handleDeeplink - it should return DeeplinkResult.ShowMyMealsPage`

Reply:
Good suggestion. But I used that naming from my current project. We're trying to use **TDD's** approach to test naming.

Your approach is similar to mine. **With** instead of **Given**, **On** instead of **When**, and **Then** is just omitted.